Support for TypeScript 2.6.x

Currently b-t-c supports  TypeScript ~2.5.2. The latest version of TS is 2.6.1. 
Upgrading to this version is a breaking change but one that would need to be supported soon to take advantage of new features like strict parameter contravariance with `strictFunctionTypes`, `// @ts-ignore` pragma, etc

Test's pass, the error encountered when running `npm run-script build` is:
```
lib/compiler.ts(116,35): error TS2345: Argument of type 'ReadonlyArray<Diagnostic>' is not assignable to parameter of type 'Diagnostic | Diagnostic[] | undefined'.
  Type 'ReadonlyArray<Diagnostic>' is not assignable to type 'Diagnostic[]'.
    Property 'push' is missing in type 'ReadonlyArray<Diagnostic>'.
lib/compiler/output-patcher.ts(51,53): error TS2345: Argument of type '(a: Entry, b: Entry) => boolean' is not assignable to parameter of type '((a: Entry, b: Entry) => {}) | undefined'.
  Type '(a: Entry, b: Entry) => boolean' is not assignable to type '(a: Entry, b: Entry) => {}'.
    Types of parameters 'a' and 'a' are incompatible.
      Type 'WalkSync.Entry' is not assignable to type 'Entry'.
        Property 'checksum' is missing in type 'Entry'.
```

--- a/lib/helpers.ts
+++ b/lib/helpers.ts
@@ -47,6 +47,7 @@ export namespace WalkSync {
     relativePath: string;
     basePath: string;
     fullPath: string;
+    checksum: string;
     mode: number;
     size: number;
     mtime: Date;

--- a/lib/interfaces.ts
+++ b/lib/interfaces.ts
@@ -22,7 +22,7 @@ export interface DiagnosticsHandler {
    *
    * Returns true if there are errors.
    */
-  check(diagnostics: Diagnostic[] | Diagnostic | undefined, throwOnError?: boolean): boolean;
+  check(diagnostics: ReadonlyArray<Diagnostic> | Diagnostic | undefined, throwOnError?: boolean): boolean;
 }
 
 export interface TypeScriptPluginOptions {

--- a/package.json
+++ b/package.json
@@ -48,7 +48,7 @@
     "fs-tree-diff": "^0.5.2",
     "heimdalljs": "0.3.3",
     "md5-hex": "^2.0.0",
-    "typescript": "~2.5.2",
+    "typescript": "~2.6.1",
     "walk-sync": "^0.3.2"
   },
   "devDependencies": {

